src/main/kotlin/net/atos/zac/smartdocuments/rest/RESTMappedSmartDocumentsTemplateGroup.kt

```diff
@@ -0,0 +1,35 @@
+/*
+ * SPDX-FileCopyrightText: 2024 Lifely
+ * SPDX-License-Identifier: EUPL-1.2+
+ */
+
+package net.atos.zac.smartdocuments.rest
+
+import nl.lifely.zac.util.AllOpen
+import nl.lifely.zac.util.NoArgConstructor
+
+@NoArgConstructor
+@AllOpen
+data class RESTMappedSmartDocumentsTemplateGroup(
+    var id: String,
+    var name: String,
+    var groups: Set<RESTMappedSmartDocumentsTemplateGroup>?,
+    var templates: Set<RESTMappedSmartDocumentsTemplate>?,
+)
+
+fun Set<RESTMappedSmartDocumentsTemplateGroup>.toStringRepresentation(): Set<String> {
+    val result = mutableSetOf<String>()
```

Comment:
I don't think you need to use a `mutableSet` here at all? Instead of a `forEach` you could use a map and then a `toSet()`? Should be much simpler code.